### PR TITLE
Adjust cpp torch trt logging level with compiler option

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

Log level of python and native TRT logger is set differently. Same level from compiler option should be helpful to work with development version of torch-TRT module.
- python trt logger: when compiler option provide debug=True
- c++ trt logger: compiled at build time

> TorchTRTLogger.cpp
#ifndef NDEBUG
  static TorchTRTLogger global_logger("[Torch-TensorRT - Debug Build] - ", LogLevel::kDEBUG, true);
#else
  static TorchTRTLogger global_logger("[Torch-TensorRT] - ", LogLevel::kWARNING, false);
#endif

And profile option for c++ trt runtime module is enabled by debug level logging. This is helpful to measure/evaluate
performance with development version of torch-TRT module

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: core/util/logging/TorchTRTLogger.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ namespace {
 
 TorchTRTLogger& get_global_logger() {
 #ifndef NDEBUG
-  static TorchTRTLogger global_logger("[Torch-TensorRT - Debug Build] - ", LogLevel::kDEBUG, true);
+  static TorchTRTLogger global_logger("[Torch-TensorRT - Debug Build] - ", LogLevel::kINFO, true);
```
Comment (narendasan) [inline core/util/logging/TorchTRTLogger.cpp:128]: I think this should still be debug

File: core/runtime/TRTEngine.cpp [hunk 2]
```diff
@@ -201,9 +201,10 @@ TRTEngine::TRTEngine(
     }
     num_io = std::make_pair(inputs_size, outputs);
   }
-
 #ifndef NDEBUG
-  this->enable_profiling();
+  if (util::logging::get_logger().get_reportable_log_level() == util::logging::LogLevel::kDEBUG) {
```
Comment (narendasan) [inline core/runtime/TRTEngine.cpp:205]: Could just remove this if you feel its not necessary

Comment (keehyuna) [reply core/runtime/TRTEngine.cpp]: Ok, I removed

File: py/torch_tensorrt/dynamo/utils.py [hunk 3]
```diff
@@ -204,10 +204,26 @@ def set_log_level(parent_logger: Any, level: Any) -> None:
...
+    elif level == logging.WARNING:
+        log_level = trt.ILogger.Severity.WARNING
+    elif level == logging.ERROR:
+        log_level = trt.ILogger.Severity.ERROR
+    elif level == logging.CRITICAL:
+        log_level = trt.ILogger.Severity.INTERNAL_ERROR
+    else:
+        raise AssertionError(f"{level} is not valid log level")
+
+    torch.ops.tensorrt.set_logging_level(int(log_level))
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/utils.py:225]: What happens in the case that it is a python only build? You might want to use enabled features to gate access to this code to only when the C++ runtime is available.

Comment (keehyuna) [reply py/torch_tensorrt/dynamo/utils.py]: You're right, I see the problem. Updated the change.
```
AttributeError: '_OpNamespace' 'tensorrt' object has no attribute 'set_logging_level'
```

**Review by narendasan — APPROVED** [review]
LGTM

**Pull request merged**